Microsoft.Quantum.Canon/StatePreparation/StatePreparation.qs

```diff
@@ -0,0 +1,208 @@
+    {
+        body
+        {
+            let maxCoefficients = 2^(Length(qubits));
+
+            if(Length(coefficients) > maxCoefficients){
+                fail "Number of coefficients must be less than or equal to 2 ^ number of control.";
+            }
+
+            // zero padding of coefficients to maxCoefficients elements.
+            let padZeros = new ComplexPolar[maxCoefficients - Length(coefficients)];
+            
+            let target = qubits[Length(qubits)-1];
+
+            let op = (Adjoint StatePreparationSBM_(coefficients + padZeros, _, _))(_, target);
+
+            if(Length(qubits) > 1){
+                let control = BigEndian(qubits[0..Length(qubits)-2]);
+                op(control);
+            }
+            else{
+                let control = BigEndian(new Qubit[0]);
+                op(control);
+            }
+
+        }
+        adjoint auto
+        controlled auto
+        adjoint controlled auto
+    }
+
+    /// # Summary
+    /// Implementation step of arbitrary state preparation procedure.
+    ///
+    /// # See Also
+    /// - Microsoft.Quantum.Canon.StatePreparationSBM
+    /// - Microsoft.Quantum.Canon.MultiplexorPauli
+    operation StatePreparationSBM_ (coefficients: ComplexPolar[], control: BigEndian, target: Qubit) : ()
+    {
+        body
+        {
+            // For each 2D block, compute disentangling single-qubit rotation parameters
+            let (disentanglingY, disentanglingZ, newCoefficients) = StatePreparationSBMComputeCoefficients_(coefficients);
+            
+            MultiplexorPauli(disentanglingZ, PauliZ, control, target);
+            MultiplexorPauli(disentanglingY, PauliY, control, target);
+            // target is now in |0> state up to the phase given by arg of newCoefficients.
+
+            // Continue recursion while there are control qubits.
+            if(Length(control) == 0){
+                let (abs, arg) = newCoefficients[0];
+                Exp([PauliI], -1.0 * arg, [target]);
+            }
+            else{
+                let newControl = BigEndian(control[0..Length(control)-2]);
+                let newTarget = control[Length(control)-1];
+                StatePreparationSBM_(newCoefficients, newControl, newTarget);
+            }
+        }   
+        adjoint auto
+        controlled auto
+        adjoint controlled auto
+    }
+
+
+    /// # Summary
+    /// Given two complex numbers $a0, a1$, computes coordinates
+    /// on the Bloch sphere such that 
+    /// $a0 \ket{0} + a1 \ket{1} = r e^{it}(e^{-i \phi /2}\cos{(\theta/2)}\ket{0}+e^{i \phi /2}\sin{(\theta/2)}\ket{1})$.
+    ///
+    /// # Input
+    /// ## a0
+    /// Complex coefficient of state $\ket{0}$.
+    /// ## a1
+    /// Complex coefficient of state $\ket{1}$.
+    ///
+    /// # Output
+    /// A tuple containing ((r, t), phi, theta).
+    function ComputeBlochSphereCoordinates(a0 : ComplexPolar, a1 : ComplexPolar) : (ComplexPolar, Double, Double) {
```

Comment:
This function has no side effects, so it might be good to drop the prefix `Compute`, such that it doesn't read as a verb.

Reply:
Fixed